src/plugins/data/public/search/aggs/buckets/terms_fn.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ExpressionFunctionDefinition } from '../../../../../expressions/public';
+import { AggExpressionType, AggExpressionFunctionArgs } from '../';
+
+const aggTypeName = 'terms';
+const fnName = 'aggTypeTerms';
+
+type Input = any;
+type Arguments = AggExpressionFunctionArgs<typeof aggTypeName>;
+type Output = AggExpressionType;
+type FunctionDefinition = ExpressionFunctionDefinition<typeof fnName, Input, Arguments, Output>;
+
+export const aggTypeTerms = (): FunctionDefinition => ({
+  name: fnName,
+  help: i18n.translate('data.search.aggs.function.buckets.terms.help', {
+    defaultMessage: 'Generates AggConfigJSON for a terms agg',
+  }),
+  type: 'agg_type',
+  args: {
+    id: {
+      types: ['string'],
+      help: '',
+    },
+    enabled: {
```

Comment:
do we need this ? 

in visualize editor you can disable an aggregation. however when we are generating expression for specific visualization we could just skip all the disabled aggs and not add them to expression at all.

this will however prevent us from at some point storing expression rather than aggconfigs json, so lets keep it for now.

Reply:
yeah my idea was to have the arguments for each of the expression functions match up with the serialized ConfigState that gets passed when creating a new aggconfig. I thought that would make it easier to reason about